training/src/main/scala/com/airbnb/aerosolve/training/DecisionTreeTrainer.scala

```diff
@@ -72,6 +72,15 @@ object DecisionTreeTrainer {
         .filter(x => x.contains(rankKey))
         .take(candidateSize)
 
+    val ex = examples(0)
+    val numFeatures = LinearRankerUtils.getNumFeatures(ex, rankKey)
```

Comment:
actually it's not really 'missing'. for example, all examples may have a feature called 'dim_market' in the hive row. but inherently, if there are 1000 markets, this feature will become 1000 different features, then for listing in 'San Francisco', the feature 'Tokyo' is always missing and won't be put under 'MISS' family since 'dim_market' feature is not missing.